+  location /auth.html {
+    root      /opt/stackstorm/static/webui/;
+  }
+
   location / {
     max_ranges 0;
     root      /opt/stackstorm/static/webui/;
@@ -137,5 +147,9 @@
     sendfile on;
     tcp_nopush on;
     tcp_nodelay on;
+
+    if ($http_cookie !~* "auth-token") {
+      rewrite ^(.*)$ /auth.html last;
+    }
   }